# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1235757391 18000
#      Fri Feb 27 12:56:31 2009 -0500
# Node ID 0c1a9c178fdd9e9ac8313c337b045c581545c300
# Parent  47fb5909519131a8fb6e39db304ddc451711ae57
replace last call to split with strsplit

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2009-02-27  John W. Eaton  <jwe@octave.org>
 
+	* general/num2str.m: Call strsplit instead of split.
+
 	* strings/strsplit.m: Style fixes.
 
 2009-02-27  Jaroslav Hajek  <highegg@gmail.com>
 
 	* strings/strsplit.m: Check also nargin.
 
 2009-02-26  John W. Eaton  <jwe@octave.org>
 
diff --git a/scripts/general/num2str.m b/scripts/general/num2str.m
--- a/scripts/general/num2str.m
+++ b/scripts/general/num2str.m
@@ -171,17 +171,17 @@ function retval = num2str (x, arg)
       else
 	fmt = "%11.5g";
       endif
     endif
     fmt = cstrcat (deblank (repmat (fmt, 1, columns (x))), "\n");
     nd = ndims (x);
     tmp = sprintf (fmt, permute (x, [2, 1, 3:nd]));
     tmp(length (tmp)) = "";
-    retval = strtrim (split (tmp, "\n"));
+    retval = strtrim (strsplit (tmp, "\n"));
   endif
 
 endfunction
 
 %!assert ((strcmp (num2str (123), "123") && strcmp (num2str (1.23), "1.23")));
 %!assert (num2str (123.456, 4), "123.5");
 %!assert (all (num2str ([1, 1.34; 3, 3.56], "%5.1f") == ["1.0  1.3"; "3.0  3.6"]));
 %!assert (num2str (1.234 + 27.3i), "1.234+27.3i");
